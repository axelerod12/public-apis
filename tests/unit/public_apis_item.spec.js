import { shallowMount } from '@vue/test-utils';
import FeldmPublicApisItem from '@/components/public_apis/public_apis_item.vue';

const item = {
  "API": "AdoptAPet",
  "Description": "Resource to help get pets adopted",
  "Auth": "apiKey",
  "HTTPS": true,
  "Cors": "yes",
  "Link": "https://www.adoptapet.com/public/apis/pet_list.html",
  "Category": "Animals",
}

describe('FeldmPublicApisItem', () => {
  let wrapper;

  beforeEach(async () => {
    wrapper = shallowMount(FeldmPublicApisItem, {
      propsData: {
        item,
      },
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders a row cell', () => {
    const tr = wrapper.find('tr');

    expect(tr.exists()).toBe(true);
  });

  it('renders 7 columns', () => {
    const columns = wrapper.findAll('td');

    expect(columns.length).toBe(7);
  });
});
