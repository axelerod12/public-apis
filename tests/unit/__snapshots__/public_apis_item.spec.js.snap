// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeldmPublicApisItem renders correctly 1`] = `
<tr>
  <td
    class="border p-2"
  >
    
    AdoptAPet
  
  </td>
   
  <td
    class="border p-2"
  >
    
    apiKey
  
  </td>
   
  <td
    class="border p-2"
  >
    
    Animals
  
  </td>
   
  <td
    class="border p-2"
  >
    
    yes
  
  </td>
   
  <td
    class="border p-2"
  >
    
    Resource to help get pets adopted
  
  </td>
   
  <td
    class="border p-2"
  >
    
    true
  
  </td>
   
  <td
    class="border p-2 underline"
  >
    <a
      href="https://www.adoptapet.com/public/apis/pet_list.html"
      target="_blank"
    >
      
      https://www.adoptapet.com/public/apis/pet_list.html
    
    </a>
  </td>
</tr>
`;
